### PR TITLE
BWS-100: + dateFormat

### PR DESCRIPTION
**Pull request by ania-leichenko**
Screenshot_3

### CODE REVIEW

File: frontend/src/components/eam/components/groups-table/helpers/date-format.helper.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const dateFormat = (dateStr: string): string => {
```
Comment (Donymak) [inline frontend/src/components/eam/components/groups-table/helpers/date-format.helper.ts]: formatDate or getFormattedDate
[B8 criteria](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/source/javascript.md#naming)

Comment (ania-leichenko) [reply frontend/src/components/eam/components/groups-table/helpers/date-format.helper.ts]: okey, thans

**Review by what1s1ove — CHANGES_REQUESTED** [review]
create a helper `getFormattedDate` in the `shared` module
```tsx
// DateFormat should be an enum

getFormattedDate(date: string, dateFormat: DateFormat): string
```
use library for dates (we already have in the `shared` module) to format the date

**Review by what1s1ove — CHANGES_REQUESTED** [review]
check the comment above

File: shared/src/helpers/date/format-date/format-date.helper.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const formatDate = (dateStr: string): string => {
```
Comment (Donymak) [inline shared/src/helpers/date/format-date/format-date.helper.ts]:
```suggestion
const getFormattedDate = (dateStr: string): string => {
```

Comment (what1s1ove) [reply shared/src/helpers/date/format-date/format-date.helper.ts]:
```tsx
// DateFormat should be an enum

getFormattedDate(date: string, dateFormat: DateFormat): string
```
use library for dates `date-fns` package to format the date

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/helpers/formatDate/formatDate.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export { formatDate } from 'bws-shared/helpers/helpers';
```
Comment (what1s1ove) [inline frontend/src/helpers/formatDate/formatDate.ts]: move to `frontend/src/helpers/date/date.ts`

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: shared/src/common/enums/date/date.enum.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+enum DateFormat {
```
Comment (what1s1ove) [inline shared/src/common/enums/date/date.enum.ts]: move to `shared/src/common/enums/date/date-format.enum.ts`

File: shared/src/helpers/date/get-formatted-date/get-formatted-date.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+const getFormattedDate = (
+  date: Date,
+  format: DateFormat = DateFormat.ISO,
+): string => {
+  if (format !== DateFormat.DISTANCE) {
+    return formatDistance(date, new Date(), {
+      addSuffix: true,
+    });
+  }
+  return formatISO(date);
```
Comment (what1s1ove) [inline shared/src/helpers/date/get-formatted-date/get-formatted-date.ts]:
```suggestion
 return format(date, formatType);
```

File: shared/src/helpers/date/get-formatted-date/get-formatted-date.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { formatDistance, formatISO } from 'date-fns';
```
Comment (what1s1ove) [inline shared/src/helpers/date/get-formatted-date/get-formatted-date.ts]:
```suggestion
import { format } from 'date-fns';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**